app/uk/gov/hmrc/transitmovementsrouter/controllers/SdesResponseParser.scala

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.transitmovementsrouter.controllers
+
+import play.api.Logging
+import play.api.libs.json.JsValue
+import play.api.mvc.BaseController
+import uk.gov.hmrc.transitmovementsrouter.controllers.errors.PresentationError
+import uk.gov.hmrc.transitmovementsrouter.models.sdes.SdesNotificationItem
+
+trait SdesResponseParser {
+  self: BaseController with Logging =>
+
+  def parseAndLogSdesResponse(responseBody: JsValue): Either[PresentationError, SdesNotificationItem] =
```

Comment:
Please use EitherT[Future, PresentationError, SdesNotificationItem] as this is going to be async call. so better to use EitherT with Future